File: 1642-water-bottles/1642-water-bottles.py
class Solution(object):
    def numWaterBottles(self, numBottles, numExchange):
        ans = numBottles
        while numBottles >= numExchange:
            rem = numBottles % numExchange
            numBottles /= numExchange
            ans += numBottles
            numBottles += rem
        
        return ans
        

File: 1642-water-bottles/1642-water-bottles.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
var numWaterBottles = function(numBottles, numExchange) {
    let ans = numBottles;
    while(numBottles >= numExchange) {
        let rem = numBottles % numExchange;
        numBottles = Math.floor(numBottles / numExchange);
        ans += numBottles;
        numBottles += rem;
    }

    return ans;
};